File: apps/electorium_web/assets/src/models/Student.js
import { Model } from '@vuex-orm/core';

export default class Student extends Model {

    static entity  = 'students';

    static fields () {

        return {

            id: this.uid(),
            publicKey: this.attr(''),
            secretKey: this.attr(''),
            name: this.attr(''),
            last_name: this.attr(''),
            email: this.attr(''),
            course: this.attr(''),
            isCandidate: this.attr(false),
            votedAt: this.number(0),
            //vote: this.attr('')
        };
    };
};

File: apps/electorium/lib/electorium/vote/student.ex
defmodule Electorium.Vote.Student do
  use Ecto.Schema
  import Ecto.Changeset


  schema "students" do
    field :publicKey, :string
    field :secretKey, :string
    field :name, :string
    field :last_name, :string
    field :email, :string
    field :course, :string
    field :isCandidate, :boolean

    timestamps()
  end

  @doc false
  def changeset(student, attrs \\ %{}) do
    student
    |> cast(attrs, [:publicKey,
                   :privateKey,
                   :name,
                   :last_name,
                   :email,
                   :course,
                   :isCandidate])
    |> validate_required([:name,
                         :email,
                         :publicKey,
                         :secretKey])
  end
end
